src/components/navbars/DesktopNav.jsx

```diff
@@ -13,24 +14,33 @@ const DesktopNav = ({
   isJoin,
 }) => {
   const { isLoggedIn } = useAuth();
-  // const {
-  //   classes,
-  //   currentPage,
-  //   totalPages,
-  //   handleSearch,
-  //   handlePageChange,
-  //   setCategory,
-  // } = useSearch();
+  const [searchTerm, setSearchTerm] = useState('');
+  const navigate = useNavigate();
+
+  const handleSearch = (event) => {
+    event.preventDefault();
+    if (searchTerm.trim()) {
+      navigate(`/search?page=1&query=${searchTerm}`);
+      setSearchTerm('');
+    }
+  };
+
   return (
     <nav
       aria-label="desktop navbar"
       className="w-2/3 hidden relative py-6 sm:flex justify-between items-center"
     >
       {isSearch ? (
-        <form className="border-2 border-red rounded-full h-10 w-2/4 flex justify-between items-center px-4 ml-6 gap-1">
+        <form
+          className="border-2 border-red rounded-full h-10 w-2/4 flex justify-between items-center px-4 ml-6 gap-1"
+          onSubmit={handleSearch}
+        >
           <input
             className="focus:outline-none w-2/3"
             aria-label="Search"
+            type="text"
+            value={searchTerm}
+            onChange={(e) => setSearchTerm(e.target.value)}
             placeholder="Search classes or teachers..."
           />
           <button onClick={onSearch} aria-label="Search">
```

Comment:
Do we need another function that triggers a redirect? `onSubmit` already handles redirecting after user submits search.

Reply:
I think we don't. Thank you Joannes!